### PR TITLE
Link through the redirector when we don't import the target

### LINKED ISSUE #11894: addons/dns/README.md 404 error
There appears to be a broken link or missing page to: http://kubernetes.io/v1.0/cluster/addons/dns/README.md
to http://kubernetes.io/v1.0/

Dead link origin page: http://kubernetes.io/v1.0/docs/getting-started-guides/mesos.html

### LINKED ISSUE #12681: High-Availability Examples Page Returns 404 Page Not Found
Error to Reproduce

Got to http://kubernetes.io/v1.0/docs/admin/high-availability.html

Click on the here button in the Vagrant Section

Link: http://kubernetes.io/v1.0/examples/high-availability/

### PR DESCRIPTION
**Pull request by krousey**
This adds some more processing to the import tool to detect when relative links escape the directories we import and link them back to github through the redirector.

fixes #11894 fixes #12681

### CODE REVIEW

**Comment by krousey** [comment]
cc @RichieEscarez

File: _tools/release_docs/main.go [hunk 1]
```diff
@@ -82,33 +131,50 @@ func processFile(prefix, filename string) error {
...
 )
 
-func rewriteLinks(fileBytes []byte) []byte {
-	return linkRE.ReplaceAllFunc(fileBytes, func(in []byte) (out []byte) {
-		match := linkRE.FindSubmatch(in)
-
-		visibleText := string(match[1])
-		linkText := string(match[2])
-		altText := ""
+func rewriteLinks(filename string, fileBytes []byte) []byte {
```
Comment (krousey) [inline _tools/release_docs/main.go]: This was mostly just refactored for reuse with `refRE`. We had some mesos documentation using reference style links that needed transforming.

**Comment by bgrant0607** [comment]
Seems like a good idea to me. 

cc @JanetKuo to review the Go code.

File: _tools/release_docs/main.go [hunk 2]
```diff
@@ -42,8 +42,57 @@ var (
...
+		// If there's no README.md in the directory, redirect to github
+		// for the directory view.
+		files, err := filepath.Glob(target + "/*")
+		if err != nil {
+			return false
+		}
+
+		hasReadme := false
+		for _, f := range files {
+			fmt.Println("KRIS:", f)
```
Comment (thockin) [inline _tools/release_docs/main.go]: debug

File: v1.0/docs/admin/high-availability.md [hunk 3]
```diff
@@ -262,7 +262,7 @@ set the `--apiserver` flag to your replicated endpoint.
 
 ## Vagrant up!
 
-We indeed have an initial proof of concept tester for this, which is available [here](../../examples/high-availability/).
+We indeed have an initial proof of concept tester for this, which is available [here](https://releases.k8s.io/v1.0.1/examples/high-availability).
```
Comment (thockin) [inline v1.0/docs/admin/high-availability.md]: Something's not right - examples IS in the gh-pages branch

Comment (krousey) [inline v1.0/docs/admin/high-availability.md]: See branch starting at line 70 in main.go.

Basically this links to a directory without a `readme.md`. This won't have an index.html when the link is followed, so you'll get a 404. In our repo, these links just take you to a directory listing. I figured getting to a version tagged github directory listing was better than 404 in these cases.

**Comment by thockin** [comment]
LGTM

**Pull request merged**